### PR TITLE
76 create function to reset db before each test suite

### LINKED ISSUE #76: Create function to reset DB before each test suite

### PR DESCRIPTION
**Pull request by Brawdunoir**

### CODE REVIEW

**Review by vazanoir — COMMENTED** [review]
GetDB more like GetRatio

File: utils/tests/tests.go [hunk 1]
```diff
@@ -93,6 +95,26 @@ func GetBasicAuthHeader(id, password string) []Header {
 		{Key: "Authorization", Value: "Basic " + authHeader}}
 }
 
+// ResetTable reset a table in the database .
+func ResetTable(db *gorm.DB, table ...interface{}) error {
+	// Foreach table, delete it.
```
Comment (WhiteDiamondz) [inline utils/tests/tests.go]:
```suggestion
	// For each table, delete it.
```

File: utils/tests/tests.go [hunk 2]
```diff
@@ -93,6 +95,26 @@ func GetBasicAuthHeader(id, password string) []Header {
 		{Key: "Authorization", Value: "Basic " + authHeader}}
 }
 
+// ResetTable reset a table in the database .
```
Comment (WhiteDiamondz) [inline utils/tests/tests.go]:
```suggestion
// ResetTable resets a table in the database.
```

File: tests/room_test.go [hunk 3]
```diff
@@ -28,6 +29,8 @@ var roomCreateRequest = utils.Request{Method: http.MethodPost, Target: roomURL,
 
 // TestCreateRoom tests the CreateRoom function.
 func TestCreateRoom(t *testing.T) {
+	utils.ResetTable(database.DB, &models.User{}, &models.Room{})
```
Comment (WhiteDiamondz) [inline tests/room_test.go]: Error return value of `utils.ResetTable` is not checked (errcheck)

File: tests/room_test.go [hunk 4]
```diff
@@ -58,6 +61,8 @@ func TestCreateRoom(t *testing.T) {
 
 // TestGetRoom tests the GetRoom function.
 func TestGetRoom(t *testing.T) {
+	utils.ResetTable(database.DB, &models.User{}, &models.Room{})
```
Comment (WhiteDiamondz) [inline tests/room_test.go]: Error return value of `utils.ResetTable` is not checked (errcheck)

File: tests/room_test.go [hunk 5]
```diff
@@ -80,6 +85,8 @@ func TestGetRoom(t *testing.T) {
 
 // TestUpdateRoom tests the UpdateRoom function.
 func TestUpdateRoom(t *testing.T) {
+	utils.ResetTable(database.DB, &models.User{}, &models.Room{})
```
Comment (WhiteDiamondz) [inline tests/room_test.go]: Error return value of `utils.ResetTable` is not checked (errcheck)

**Review by WhiteDiamondz — CHANGES_REQUESTED** [review]
Small changes required

File: database/db.go [hunk 6]
```diff
@@ -11,17 +11,16 @@ import (
 	"gorm.io/gorm/logger"
 )
 
-var DB *gorm.DB = GetDatabase()
+var DB *gorm.DB = setupDatabase()
 
-// GetDatabase returns a database instance
-func GetDatabase() *gorm.DB {
+// setupDatabase returns a setuped database instance
```
Comment (WhiteDiamondz) [inline database/db.go]:
```suggestion
// setupDatabase returns a setuped database instance.
```

File: routes/router.go [hunk 7]
```diff
@@ -10,15 +10,18 @@ import (
...
 	"github.com/Brawdunoir/dionysos-server/variables"
 	cache "github.com/chenyahui/gin-cache"
 	"github.com/chenyahui/gin-cache/persist"
 	"github.com/gin-gonic/gin"
 	"github.com/go-redis/redis/v8"
+	"gorm.io/gorm"
 )
 
 var redisStore *persist.RedisStore
+var db *gorm.DB = database.DB
```
Comment (WhiteDiamondz) [inline routes/router.go:25]:
```suggestion
// Database pointer that will be used in the routes.
var db *gorm.DB = database.DB
```

File: utils/tests/tests.go [hunk 8]
```diff
@@ -95,9 +95,9 @@ func GetBasicAuthHeader(id, password string) []Header {
 		{Key: "Authorization", Value: "Basic " + authHeader}}
 }
 
-// ResetTable reset a table in the database .
+// ResetTable resets a table in the database.
 func ResetTable(db *gorm.DB, table ...interface{}) error {
-	// Foreach table, delete it.
+	// For each table, delete it.
```
Comment (WhiteDiamondz) [inline utils/tests/tests.go]:
```suggestion
	// Drop all tables.
```

**Review by WhiteDiamondz — CHANGES_REQUESTED** [review]

**Review by WhiteDiamondz — APPROVED** [review]

**Pull request merged**